### PR TITLE
feat: create new category and task

### LINKED ISSUE #199: Create new category and task
## Description

Implement logic to create new share house on the new page at the same time.

## Criteria

- [ ] Create a new record on `category` table
- [ ] Create a new record on `task` table

### PR DESCRIPTION
**Pull request by KaiKoide**
## Overview

Implement logic to create a new share house on the new page at the same time.

## Changes

- Create a new record on `category` table
- Create a new record on `task` table

## Review points

1. Set Content-Type: application/json on Headers tab in Postman
2. Send a POST method with the following data after selecting raw on Body tab
```
{
    "category": "test-category",
    "title": "test-title",
    "description": "test-description"
}
```
3. Check whether a new record is added to `tenant` table and `task` table

## Assignee Checklist:

- [ ] The base branch is correct (no accidental merges)
- [ ] The branch name follows our branch naming rules
- [ ] The PR title follows our PR title rules
- [ ] My code follows our coding style
- [ ] I have commented on my code, particularly in hard-to-understand areas
- [ ] My changes generate no new warnings or errors in the console

## Reviewer Checklist:

- [x] Code readability and simplicity
- [x] Follows best practices and coding standards
- [x] Understandable and maintainable code

### CODE REVIEW

File: src/app/api/[[...route]]/routes/category.route.ts
Comment (akiomatic) [inline src/app/api/[[...route]]/routes/category.route.ts:1]: Could you add a check whether the number of existing categories already reaches the max or not before creating one?

https://github.com/Tascurator/tascurator-documents/blob/main/docs/docs/modeling.md#category---%E3%82%AB%E3%83%86%E3%82%B4%E3%83%AA%E3%83%BC

**Review by akiomatic — CHANGES_REQUESTED** [review]
Thank you. I'd like you to check my comments.

File: src/app/api/[[...route]]/routes/category.route.ts
Comment (KaiKoide) [reply src/app/api/[[...route]]/routes/category.route.ts:1]: I added an if statement for that.

**Review by akiomatic — APPROVED** [review]
LGTM. Thank you.

**Pull request merged**